### DISCUSSION TITLE
Как собрать этот проект у себя со статической линковкой?

### DISCUSSION THREAD

**Discussion by Vladimir-N-sk** [hufrea/byedpi#116] [General]
Извините, если пишу не туда, я новичок на github.

Склонировал ваш проект себе на комп. С помошью make и вашего Makefile все пркрасно компилируется и линкуется, но программа получается с динамической линковкой. 
Релиз программы скаченной отсюда - статический. Как? Может какие доп.опции компилятору указать?

Да, ОС - Ubuntu 18.04

**Comment by zimbabwe303** [comment]
Теоретически: `LDFLAGS="-static" make`

**Reply by Vladimir-N-sk** [reply]
Да вот что-то как-то не прокатило.

cc -o ciadpi packets.o main.o conev.o proxy.o desync.o mpool.o extend.o -static
main.o: In function `get_addr':
main.c:(.text+0x4be): warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking

**Reply by zimbabwe303** [reply]
Это только warning. Он собрался, скорее всего. А как glibc полностью прилинковать не знаю. Поищите, где-то на stackoverflow.com попадалось вроде.

**Reply by Vladimir-N-sk** [reply]
Действительно. Вот я тупик. 
Спасибо.

**Reply by zimbabwe303** [reply]
getaddrinfo находится в libnss, glibc с ним слинкован только динамически, зато musl может и статически:
`apt-get install musl musl-tools`

Затем для сборки:
`CC="musl-gcc" CFLAGS="-idirafter /usr/include -idirafter /usr/include/x86_64-linux-gnu" LDFLAGS="-static" make`
(только сначала не забыть сделать `make clean`, конечно)

Проверить, что всё собралось действительно статически можно при помощи `ldd ciadpi`

**Reply by Vladimir-N-sk** [reply]
Да вроде все собралось и работает без танцев с бубном, родным gcc. Только бинарник получился на порядок больше чем у автора, но это не страшно.

mon@XYZ:~/lib$ file ciadpi
ciadpi: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), statically linked, for GNU/Linux 3.2.0,

mon@XYZ:~/lib$ ldd ciadpi
        не является динамическим исполняемым файлом
